### ISSUE TITLE
`ERROR: column "..." being dropped, try again later` when applying migrations with CRDB 23.1

### ISSUE THREAD

**Issue by aqrln** [prisma/prisma#20851]
### Bug description

When the same column has to be dropped and re-added in a migration, this fails in CockroachDB 23.1 with the following error:

```
Error: P3018

A migration failed to apply. New migrations cannot be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve

Migration name: 20230825172551_sec

Database error code: 55000

Database error:
ERROR: column "num" being dropped, try again later

DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState(E55000), message: "column \"num\" being dropped, try again later", detail: None, hint: None, position: None, where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("add_column.go"), line: Some(244), routine: Some("checkColumnDoesNotExist") }
```

This was discovered in sql-migration-tests when adding CockroachDB 23.1 to CI in https://github.com/prisma/prisma-engines/pull/4132. Looks like an upstream bug that needs to be reported to CockroachDB.

### How to reproduce

1. Use the schema below
2. `prisma migrate dev`
3. Change `Decimal` to `Boolean`
4. `prisma migrate dev`

### Expected behavior

The column type should have been changed successfully.

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "cockroachdb"
  url = env("DATABASE_URL")
}

model A {
  id  Int      @id @default(sequence())
  num Decimal
}
```

### Environment & setup

- OS: macOS
- Database: CockroachDB 23.1
- Node.js version: v20.5.1

### Prisma Version

```
prisma                  : 0.0.0
@prisma/client          : 0.0.0
Current platform        : darwin-arm64
Query Engine (Node-API) : E_CANNOT_RESOLVE_VERSION (at ../../../prisma-engines/target/debug/libquery_engine.dylib, resolved by PRISMA_QUERY_ENGINE_LIBRARY)
Schema Engine           : schema-engine-cli c2478af070f89607a31a77361e0053fd3f0ad1d4 (at ../../../prisma-engines/target/debug/schema-engine, resolved by PRISMA_SCHEMA_ENGINE_BINARY)
Schema Wasm             : @prisma/prisma-schema-wasm 5.3.0-5.018067b0442e15ab541fc2d442e5ffdc1f3a051b
Default Engines Hash    : 018067b0442e15ab541fc2d442e5ffdc1f3a051b
Studio                  : 0.494.0
Preview Features        : tracing
```
5.3.0-dev

**Comment by Jolg42** [comment]
It looks like a user reported this 2 weeks ago too
https://github.com/prisma/prisma/issues/20707

**Comment by Jolg42** [comment]
I found this issue on CockroachDB's GitHub which look similar but is about indexes
https://github.com/cockroachdb/cockroach/issues/98494

**Comment by nickelnickel** [comment]
I believe this needs the same workaround as v22.2, as described here: https://github.com/prisma/prisma-engines/pull/3742

The specific commit that applies to this issue (for 22.2) is: https://github.com/prisma/prisma-engines/pull/3742/commits/e8f9f515e88357f9efb3b9064b1491db3631ab53

This will `SET enable_implicit_transaction_for_batch_statements=false` during prisma migrations and should avoid the issue. The workaround looks for 22.2, but does not have 23.1 which requires the same workaround.

**Comment by Jolg42** [comment]
Note I tried locally using CRDB `v23.2.0-rc.2` and `v23.1.14` and can confirm it's fixed 💯

**Comment by Jolg42** [comment]
For the fix, use Prisma `5.9.0` (will be released today)
